### PR TITLE
V16

### PR DESCRIPTION
**Pull request by noranda**
This PR is to merge v16 into main in preparation for the Recipe 16.0.0 release. Branches are in sync.

### CODE REVIEW

File: packages/recipe/src/components/EzFormGroup/EzFormGroup.types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {ReactNode} from 'react';
+
+export type Ref = HTMLDivElement;
+
+export interface EzFormGroupProps {
+  ariaLabel?: string;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzFormGroup/EzFormGroup.types.ts:6]: ❓ why is aria label optional? should it be required?

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+) & {
+  /** @deprecated */
+  className?: string;
+  /** @deprecated */
+  label?: string;
+  onChange?: (event: ChangeEvent<HTMLInputElement>, checked: boolean) => void;
+};
+
+export type EzCheckboxMuiProps = EzCheckboxCommonProps & {
+  ariaLabel?: string;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:39]: same question about optional aria label

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  color?: EzThemeColors;
+  defaultChecked?: boolean;
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:42]: does it always need an onChange?

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  size?: string;
+  value?: any;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:44]: does it always need a value?

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+export type EzCheckboxLegacyAcknowledgementProps = EzCheckboxCommonProps & {
+  /** @deprecated */
+  acknowledgement: true;
+  /** @deprecated */
+  terms: string;
+};
+
+export type EzCheckboxLegacyClassicProps = EzCheckboxCommonProps & {
+  acknowledgement?: boolean;
+  terms?: undefined;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:24]: why undefined here?

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import {ChangeEvent, FocusEventHandler} from 'react';
+import {type EzThemeColors} from '../../themes';
+
+export type Ref = HTMLInputElement;
+
+export type EzCheckboxCommonProps = {
+  id?: string;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:7]: does it always need an id?

**Comment by danidewitt** [comment]
👍 looks great, as it should since this was all code reviewed before. i had a couple props questions after re-reviewing, but nothing major.

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts (hunk 2 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:39]: If the checkbox has a label, then `aria-label` isn't required.

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts (hunk 3 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:42]: It only needs an `onChange` if it's controlled.

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts (hunk 4 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:44]: `value` defaults to "on" ([source](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value))

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts (hunk 5 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:24]: In this case it's similar to `never` and was needed to type narrow it from the union type. Typescript is weird!

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts (hunk 6 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/EzCheckbox.types.ts:7]: As long as the checkbox is wrapped inside the label (as ours are), the `id` is optional.

```js
// id optional
<label>
  <input type="checkbox">
  text
</label>

// id required
<label htmlFor="cb-1">text</label>
<input type="checkbox" id="cb-1">
```

**Review by pgrippi — APPROVED** [review]

**Review by danidewitt — APPROVED** [review]

**Pull request merged**